src/ArborX_DBSCAN.hpp

```diff
@@ -203,12 +205,26 @@ dbscan(ExecutionSpace const &exec_space, Primitives const &primitives,
 
                          Kokkos::atomic_fetch_add(&cluster_sizes(labels(i)), 1);
                        });
-  Kokkos::parallel_for("ArborX::dbscan::mark_noise",
-                       Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, n),
-                       KOKKOS_LAMBDA(int const i) {
-                         if (cluster_sizes(labels(i)) == 1)
-                           labels(i) = -1;
-                       });
+  if (is_special_case)
+  {
+    // Cannot use CCSCore as it always returns true
+    Kokkos::parallel_for("ArborX::dbscan::mark_noise",
+                         Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, n),
+                         KOKKOS_LAMBDA(int const i) {
+                           if (cluster_sizes(labels(i)) == 1)
+                             labels(i) = -1;
+                         });
+  }
+  else
+  {
+    DBSCAN::DBSCANCorePoints<MemorySpace> is_core{num_neigh, core_min_size};
+    Kokkos::parallel_for("ArborX::dbscan::mark_noise",
+                         Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, n),
+                         KOKKOS_LAMBDA(int const i) {
+                           if (cluster_sizes(labels(i)) == 1 && !is_core(i))
+                             labels(i) = -1;
+                         });
+  }
```

Comment:
I reverted that patch out of curiosity and it looks like the unit tests are still passing.

Reply:
Yeah, I did the same. I think some permutation of the points in the test may trigger the failure in Serial, but it's really hard to get to it.